src/app/server/RendezvousServer.h

```diff
@@ -31,6 +32,7 @@ class RendezvousServer : public RendezvousSessionDelegate
     CHIP_ERROR Init(const RendezvousParameters & params, TransportMgrBase * transportMgr, SecureSessionMgr * sessionMgr,
                     Transport::AdminPairingInfo * admin);
     void SetDelegate(AppDelegate * delegate) { mDelegate = delegate; };
+    void SetStorage(PersistentStorageDelegate * delegate) { mStorage = delegate; };
```

Comment:
Why are these delegates not set at constructor/Init time?

Reply:
I don't think there's a good reason. I'll refactor it to move to Init.

Reply:
Actually, looking at the code, `Init` is opening the pairing window. So it could be called multiple times during the uptime of a device (i.e. whenever the pairing window is opened). SetDelegate/SetStorage is called only at the startup. I will combine the SetDelegate/SetStorage into one function, SetDelegates(). Init may be renamed to something better. Maybe `WaitForPairing`?

Reply:
Agree - Init should be a 'one time initialization with everything required'. I would expect some mandatory delegates (like storage callbacks) to be set during init.

Reply:
Renamed SetDelegates to Init. Also added a check for storage delegate.